core/consensus/babe/impl/babe_impl.hpp

```diff
@@ -186,6 +199,19 @@ namespace kagome::consensus::babe {
         clock::SystemClock::TimePoint slot_timestamp,
         std::optional<std::reference_wrapper<const crypto::VRFOutput>> output,
         primitives::AuthorityIndex authority_index);
+    /**
+     * `processSlotLeadership` coroutine piece
```

Comment:
This can be confusing because it's not actually a coroutine, I'd rather that you describe it in the comment the way it is.